apps/api/src/domain/models/tag/commands/tag-resource-or-note/resource-or-note-tagged.event-handler.spec.ts

```diff
@@ -0,0 +1,225 @@
+import { Environment } from '../../../../../app/config/constants/environment';
+import { buildMultilingualTextWithSingleItem } from '../../../../../domain/common/build-multilingual-text-with-single-item';
+import { MultilingualText } from '../../../../../domain/common/entities/multilingual-text';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { Maybe } from '../../../../../lib/types/maybe';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { ArangoConnectionProvider } from '../../../../../persistence/database/arango-connection.provider';
+import { ArangoDatabase } from '../../../../../persistence/database/arango-database';
+import { ArangoDatabaseForCollection } from '../../../../../persistence/database/arango-database-for-collection';
+import { ArangoDatabaseProvider } from '../../../../../persistence/database/database.provider';
+import mapDatabaseDocumentToAggregateDTO from '../../../../../persistence/database/utilities/mapDatabaseDocumentToAggregateDTO';
+import mapEntityDTOToDatabaseDocument, {
+    ArangoDatabaseDocument,
+} from '../../../../../persistence/database/utilities/mapEntityDTOToDatabaseDocument';
+import { PersistenceModule } from '../../../../../persistence/persistence.module';
+import generateDatabaseNameForTestSuite from '../../../../../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import { EventSourcedTagViewModel } from '../../../../../queries/buildViewModelForResource/viewModels/tag.view-model.event-sourced';
+import { TestEventStream } from '../../../../../test-data/events';
+import { DTO } from '../../../../../types/DTO';
+import buildDummyUuid from '../../../__tests__/utilities/buildDummyUuid';
+import { ResourceOrNoteTagged } from './resource-or-note-tagged.event';
+import { ResourceOrNoteTaggedEventHandler } from './resource-or-note-tagged.event-handler';
+
+const WIDGET_COLLECTION = 'widgets';
+
+class WidgetViewModel {
+    id: string;
+    name: MultilingualText;
+    tags: EventSourcedTagViewModel[];
+
+    constructor({ id, name, tags }: DTO<WidgetViewModel>) {
+        this.id = id;
+
+        this.name = new MultilingualText(name);
+
+        this.tags = tags.map((t) => new EventSourcedTagViewModel(t));
+    }
+}
+
+interface IWidgetQueryRepository {
+    fetchById(id: string): Promise<Maybe<WidgetViewModel>>;
+    create(w: WidgetViewModel): Promise<void>;
+    tag(id: string, tagId: string): Promise<void>;
+}
+
+const targetTagLabel = 'car parts';
+
+const targetTagId = buildDummyUuid(2);
+
+const knownTags: EventSourcedTagViewModel[] = [
+    {
+        type: AggregateType.tag,
+        id: targetTagId,
+        label: targetTagLabel,
+        members: [],
+        name: buildMultilingualTextWithSingleItem(targetTagLabel),
+    },
+].map((t) => new EventSourcedTagViewModel(t));
+
+/**
+ * We do not want to couple this test to the domain, which is bound to shift
```

Comment:
An interesting question is whether we want to just use mocks for all event handler tests. What I like about our current pattern is that it gives us complete confidence all the way to the DB and it is resilient to refactoring the way handlers interact with persistence layers if need be.